p2p/switch.go

```diff
@@ -365,10 +363,14 @@ func (sw *Switch) reconnectToPeer(addr *NetAddress) {
 		// sleep an exponentially increasing amount
 		sleepIntervalSeconds := math.Pow(reconnectBackOffBaseSeconds, float64(i))
 		sw.randomSleep(time.Duration(sleepIntervalSeconds) * time.Second)
+
 		err := sw.DialPeerWithAddress(addr, true)
 		if err == nil {
 			return // success
 		}
+		if _, ok := err.(ErrCurrentlyDialingOrExistingAddress); ok {
```

Comment:
Can we make this an `else if` ? Nice to keep the structure of the code reflect the logic